Fix IntelliSense class suggestions

thecrypticace opened the pull request:
This PR fixes a few problems with IntelliSense:
- Default installs suggest font weight utilities twice
- When setting `--font-weight-*: initial;` the default weights are still suggested
- Setting `--spacing-*: initial;` still shows suggestions for utilities using the spacing scale assuming it's multiplier-based. Removing the `--spacing: …` key should make them entirely theme driven instead.
- The default container query variants like `@sm`, `@md`, etc… aren't suggested. Only the `@min-sm` and `@max-sm` style ones are.

Fixes https://github.com/tailwindlabs/tailwindcss-intellisense/issues/1133
Fixes https://github.com/tailwindlabs/tailwindcss-intellisense/issues/1136
Fixes #15811

thecrypticace committed:
Don’t suggest “bare” values for font weight utilities

Also prevents duplicate font weight suggestions

thecrypticace committed:
Don't suggest bare values when `--spacing: …` isn't used

thecrypticace committed:
Update tests

thecrypticace committed:
Suggest container variants that don’t use `@min-…`

thecrypticace committed:
Update snapshots

thecrypticace committed:
Merge branch 'next' into fix/v4-incorrect-suggestions

thecrypticace committed:
Update changelog

florianbouvot commented:
@thecrypticace can you reorder colors 50 before 100, 200, ...? 

![Capture d’écran 2025-01-28 à 15 19 01](https://github.com/user-attachments/assets/14321e29-38e4-4eae-b4d9-70aeebd5b021)

florianbouvot commented:
@thecrypticace I think there are the same issue with spacing

![Capture d’écran 2025-01-28 à 15 22 34](https://github.com/user-attachments/assets/10618614-136d-47a0-999c-e8968c3e0ef9)

RobinMalfait committed:
add natural sorting

This will ensure that strings that contain numbers are sorted as if a
human would sort it.

RobinMalfait committed:
update snapshot to use the natural sorted class list

RobinMalfait committed:
use existing `compare` function

RobinMalfait commented on packages/tailwindcss/src/intellisense.ts:
```diff
@@ -33,7 +34,7 @@ export function getClassList(design: DesignSystem): ClassEntry[] {
   }
 
-  list.sort((a, b) => (a[0] === b[0] ? 0 : a[0] < b[0] ? -1 : 1))
+  list.sort(([a], [b]) => compare(a, b))
```
I added this, but I wonder if it's better todo this instead:
```suggestion
  list.sort((a, b) => compare(a[0], b[0]))
```
🤔

RobinMalfait committed:
Update packages/tailwindcss/src/intellisense.ts

thecrypticace committed:
Update changelog

thecrypticace committed:
Merge branch 'next' into fix/v4-incorrect-suggestions
